docs/contributing.md

```diff
@@ -0,0 +1,136 @@
+Install the [required tools](required-tools.md).
+
+### Authentication
+Configure authentication for the cluster using one of the following options:
+
+1. Contact a member of the KubeSaw Team for instructions on how to configure the cluster to use our internal Dev SSO.
+
+2. Configure your own Keycloak server and set up authentication on the OpenShift cluster: [configuring internal oauth](https://docs.openshift.com/container-platform/4.6/authentication/configuring-internal-oauth.html)
+
+3. Deploy and configure keycloak internally as part of the cluster. Just add `DEV_SSO=true` parameter to the `dev` targets. For eg.: `make dev-deploy-latest DEV_SSO=true` will deploy latest version of the operators with a preconfigured keycloak instance and one default keycloak user `user1@user.us` with password `user1`.
+If you are presented with the following error, then you need to accept the self-signed certificate of the dev Keycloak instance first. Go to `https://keycloak-<dev-sso-namespace>.<domain>/auth (the complete link is printed out at the end of the command) and accept the certificate.:
+
+![insecure keycloak](images/insecure_keycloak.png "Insecure keycloak configuration")
+
+**NOTE**: This third option *only works with OCP and CRC* clusters atm.
+
+## Install
+
+### Remove Self Provisioner Role
+
+It is strongly recommended to remove the self-provisioner role to disallow users from creating their own namespaces. This is because KubeSaw is designed to create/manage namespaces for users automatically. It creates these namespaces based on predefined templates that also define resource limits so only these namespaces should be accessible to KubeSaw users.
+
+Run the following commands:
+```shell
+oc patch clusterrolebinding.rbac self-provisioners -p '{"subjects": null, "metadata": {"annotations":{"rbac.authorization.kubernetes.io/autoupdate": "false"}}}'
+oc adm policy remove-cluster-role-from-group self-provisioner system:authenticated:oauth
+```
+
+### Install KubeSaw
+Clone this repository `git clone git@github.com:codeready-toolchain/toolchain-e2e.git`
+
+This repository provides you multiple Makefile targets that you can use - it depends on which version of KubeSaw operators you want to install.
+
+**NOTE**: If the cluster is an OSD cluster, then set the variable `IS_OSD=true` when running any of the Makefile targets (for example: `make appstudio-dev-deploy-latest IS_OSD=true`).
+
+**IMPORTANT**: Make note of the Registration Service URL that is printed at the end of the target execution.
+
+### Latest greatest KubeSaw
+Run the following to install the latest greatest KubeSaw operators in dev mode:
+```
+make dev-deploy-latest
+```
+
+### Latest greatest KubeSaw for AppStudio
+Run the following to install the latest greatest KubeSaw operators in dev mode for AppStudio environment:
+```
+make appstudio-dev-deploy-latest
+```
+
+### Local version
+If you want to install a local version of a specific KubeSaw operator in dev mode then:
```

Comment:
This assumes certain directory layout or additional config, I think. Shouldn't we mention it here, too?

Reply:
good point, added more details in https://github.com/kubesaw/kubesaw.github.io/pull/10/commits/f7fc24c415cfc0f65cf0a928707b8fc569578319